### PR TITLE
Support upsert

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo <xinguo.li@zilliz.com>
#54

### CODE REVIEW

File: proto/milvus.proto [hunk 1]
```diff
@@ -709,16 +710,32 @@ message InsertRequest {
...
-  schema.IDs IDs = 2; // required for insert, delete
+  schema.IDs IDs = 2; // required for insert, delete, update
   repeated uint32 succ_index = 3; // error indexes indicate
   repeated uint32 err_index = 4; // error indexes indicate
   bool acknowledged = 5;
   int64 insert_cnt = 6;
   int64 delete_cnt = 7;
   int64 upsert_cnt = 8;
   uint64 timestamp = 9;
+  int64 update_cnt = 10;
```
Comment (yah01) [inline proto/milvus.proto:738]: `upsert_cnt` is actually the same as `update_cnt`

File: proto/milvus.proto [hunk 2]
```diff
@@ -712,9 +713,24 @@ message InsertRequest {
...
+message UpsertRequest {
+  option (common.privilege_ext_obj) = {
+    object_type: Collection
+    object_privilege: PrivilegeUpsert
+    object_name_index: 3
+  };
+  common.MsgBase base = 1;
+  string db_name = 2;
+  string collection_name = 3;
+  string partition_name = 4;
```
Comment (czs007) [inline proto/milvus.proto:725]: What is the semantics of upsert when the partition parameter is specified?

Comment (smellthemoon) [reply proto/milvus.proto:725]: It follows the partition semantic logic of delete and insert.

File: proto/common.proto [hunk 3]
```diff
@@ -304,22 +306,23 @@ enum ObjectPrivilege {
...
-    PrivilegeQuery = 16;
-    PrivilegeLoadBalance = 17;
-    PrivilegeImport = 18;
-    PrivilegeCreateOwnership = 19;
-    PrivilegeUpdateUser = 20;
-    PrivilegeDropOwnership = 21;
-    PrivilegeSelectOwnership = 22;
-    PrivilegeManageOwnership = 23;
-    PrivilegeSelectUser = 24;
+    PrivilegeUpsert = 10;
```
Comment (czs007) [inline proto/common.proto:309]: Define PribilegeUpsert as 25 instead of 10.
Do not modify previously defined value.

**Pull request merged**